<a href="https://colab.research.google.com/github/gcarmonaar/Diplomado-CDP-24/blob/main/DiplomadoCD_modulo2_clase2_p2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

In [ ]:
salarios=pd.read_csv("/content/salary.csv")

In [ ]:
salarios.head()

age          workclass  fnlwgt   education  education-num  \
0   39          State-gov   77516   Bachelors             13   
1   50   Self-emp-not-inc   83311   Bachelors             13   
2   38            Private  215646     HS-grad              9   
3   53            Private  234721        11th              7   
4   28            Private  338409   Bachelors             13   

        marital-status          occupation    relationship    race      sex  \
0        Never-married        Adm-clerical   Not-in-family   White     Male   
1   Married-civ-spouse     Exec-managerial         Husband   White     Male   
2             Divorced   Handlers-cleaners   Not-in-family   White     Male   
3   Married-civ-spouse   Handlers-cleaners         Husband   Black     Male   
4   Married-civ-spouse      Prof-specialty            Wife   Black   Female   

   capital-gain  capital-loss  hours-per-week  native-country  salary  
0          2174             0              40   United-States   <=50K  
1             0             0              13   United-States   <=50K  
2             0             0              40   United-States   <=50K  
3             0             0              40   United-States   <=50K  
4             0             0              40            Cuba   <=50K

In [ ]:
salarios.tail()

age      workclass  fnlwgt    education  education-num  \
32556   27        Private  257302   Assoc-acdm             12   
32557   40        Private  154374      HS-grad              9   
32558   58        Private  151910      HS-grad              9   
32559   22        Private  201490      HS-grad              9   
32560   52   Self-emp-inc  287927      HS-grad              9   

            marital-status          occupation relationship    race      sex  \
32556   Married-civ-spouse        Tech-support         Wife   White   Female   
32557   Married-civ-spouse   Machine-op-inspct      Husband   White     Male   
32558              Widowed        Adm-clerical    Unmarried   White   Female   
32559        Never-married        Adm-clerical    Own-child   White     Male   
32560   Married-civ-spouse     Exec-managerial         Wife   White   Female   

       capital-gain  capital-loss  hours-per-week  native-country  salary  
32556             0             0              38   United-States   <=50K  
32557             0             0              40   United-States    >50K  
32558             0             0              40   United-States   <=50K  
32559             0             0              20   United-States   <=50K  
32560         15024             0              40   United-States    >50K

In [ ]:
salarios.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32561 entries, 0 to 32560
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   age             32561 non-null  int64 
 1   workclass       32561 non-null  object
 2   fnlwgt          32561 non-null  int64 
 3   education       32561 non-null  object
 4   education-num   32561 non-null  int64 
 5   marital-status  32561 non-null  object
 6   occupation      32561 non-null  object
 7   relationship    32561 non-null  object
 8   race            32561 non-null  object
 9   sex             32561 non-null  object
 10  capital-gain    32561 non-null  int64 
 11  capital-loss    32561 non-null  int64 
 12  hours-per-week  32561 non-null  int64 
 13  native-country  32561 non-null  object
 14  salary          32561 non-null  object
dtypes: int64(6), object(9)
memory usage: 3.7+ MB


In [ ]:
var_cat=['salary', 'workclass', 'education','marital-status', 'occupation', 'relationship', 'race', 'sex', 'native-country'] #lista de variables que deberían tener categorías
for variable in var_cat: #para cada elemento de la lista var_cat
    salarios[variable] = salarios[variable].astype('category') #convertir la variable a categorías

In [ ]:
salarios.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32561 entries, 0 to 32560
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype   
---  ------          --------------  -----   
 0   age             32561 non-null  int64   
 1   workclass       32561 non-null  category
 2   fnlwgt          32561 non-null  int64   
 3   education       32561 non-null  category
 4   education-num   32561 non-null  int64   
 5   marital-status  32561 non-null  category
 6   occupation      32561 non-null  category
 7   relationship    32561 non-null  category
 8   race            32561 non-null  category
 9   sex             32561 non-null  category
 10  capital-gain    32561 non-null  int64   
 11  capital-loss    32561 non-null  int64   
 12  hours-per-week  32561 non-null  int64   
 13  native-country  32561 non-null  category
 14  salary          32561 non-null  category
dtypes: category(9), int64(6)
memory usage: 1.8 MB


In [ ]:
salarios.describe(include='category')

workclass education       marital-status       occupation relationship  \
count      32561     32561                32561            32561        32561   
unique         9        16                    7               15            6   
top      Private   HS-grad   Married-civ-spouse   Prof-specialty      Husband   
freq       22696     10501                14976             4140        13193   

          race    sex  native-country  salary  
count    32561  32561           32561   32561  
unique       5      2              42       2  
top      White   Male   United-States   <=50K  
freq     27816  21790           29170   24720

**Workclass** - esta variable nos indica el tipo de empresa, hay 9 tipos diferentes y la frecuencia más alta es el tipo privado con un total de 22696 registros



In [ ]:
salarios['relationship']

0         Not-in-family
1               Husband
2         Not-in-family
3               Husband
4                  Wife
              ...      
32556              Wife
32557           Husband
32558         Unmarried
32559         Own-child
32560              Wife
Name: relationship, Length: 32561, dtype: category
Categories (6, object): [' Husband', ' Not-in-family', ' Other-relative', ' Own-child',
                         ' Unmarried', ' Wife']

In [ ]:
salarios.describe()

age        fnlwgt  education-num  capital-gain  capital-loss  \
count  32561.000000  3.256100e+04   32561.000000  32561.000000  32561.000000   
mean      38.581647  1.897784e+05      10.080679   1077.648844     87.303830   
std       13.640433  1.055500e+05       2.572720   7385.292085    402.960219   
min       17.000000  1.228500e+04       1.000000      0.000000      0.000000   
25%       28.000000  1.178270e+05       9.000000      0.000000      0.000000   
50%       37.000000  1.783560e+05      10.000000      0.000000      0.000000   
75%       48.000000  2.370510e+05      12.000000      0.000000      0.000000   
max       90.000000  1.484705e+06      16.000000  99999.000000   4356.000000   

       hours-per-week  
count    32561.000000  
mean        40.437456  
std         12.347429  
min          1.000000  
25%         40.000000  
50%         40.000000  
75%         45.000000  
max         99.000000

In [ ]:
salarios["hours-per-day"]=salarios["hours-per-week"]/5

In [ ]:
salarios["hours-per-day"].head(10)

0     8.0
1     2.6
2     8.0
3     8.0
4     8.0
5     8.0
6     3.2
7     9.0
8    10.0
9     8.0
Name: hours-per-day, dtype: float64

In [ ]:
print("El número máximo de horas por semana es de: ",salarios['hours-per-week'].max())
print("Mínimo de hrs por semana:",salarios['hours-per-week'].min())
print("Varianza de hrs por semana:",salarios['hours-per-week'].var())
print("Desviación estándar de hrs por semana: ",salarios['hours-per-week'].std())
#print("Suma de hrs por semana: ",salarios['hours-per-week'].sum())
print("Promedio de hrs por semana: ",salarios['hours-per-week'].mean())


El número máximo de horas por semana es de:  99
Mínimo de hrs por semana: 1
Varianza de hrs por semana: 152.45899505045418
Desviación estándar de hrs por semana:  12.347428681731843
Promedio de hrs por semana:  40.437455852092995


In [ ]:
#Ordenar de acuerdo con la variable hours-per-week de manera ascendente
salarios.sort_values("hours-per-week", ascending=True)

age          workclass  fnlwgt      education  education-num  \
19750   23            Private   72887        HS-grad              9   
25078   74            Private  260669           10th              6   
11451   27            Private  147951        HS-grad              9   
8447    67                  ?  244122      Assoc-voc             11   
32525   81                  ?  120478      Assoc-voc             11   
...    ...                ...     ...            ...            ...   
25806   49                  ?   31478        7th-8th              4   
30610   51   Self-emp-not-inc   24790   Some-college             10   
10143   35                  ?  164866           10th              6   
4308    35   Self-emp-not-inc  166416        HS-grad              9   
1887    55   Self-emp-not-inc  184425   Some-college             10   

            marital-status          occupation     relationship  \
19750        Never-married        Craft-repair        Own-child   
25078             Divorced       Other-service    Not-in-family   
11451        Never-married   Machine-op-inspct   Other-relative   
8447               Widowed                   ?    Not-in-family   
32525             Divorced                   ?        Unmarried   
...                    ...                 ...              ...   
25806   Married-civ-spouse                   ?          Husband   
30610   Married-civ-spouse     Exec-managerial          Husband   
10143             Divorced                   ?    Not-in-family   
4308    Married-civ-spouse     Farming-fishing          Husband   
1887    Married-civ-spouse     Farming-fishing          Husband   

                      race      sex  capital-gain  capital-loss  \
19750   Asian-Pac-Islander     Male             0             0   
25078                White   Female             0             0   
11451                White     Male             0             0   
8447                 White   Female             0             0   
32525                White   Female             0             0   
...                    ...      ...           ...           ...   
25806                White     Male             0             0   
30610                White     Male             0             0   
10143                White     Male             0             0   
4308                 White     Male             0             0   
1887                 White     Male             0             0   

       hours-per-week  native-country  salary  hours-per-day  
19750               1         Vietnam   <=50K            0.2  
25078               1   United-States   <=50K            0.2  
11451               1   United-States   <=50K            0.2  
8447                1   United-States   <=50K            0.2  
32525               1               ?   <=50K            0.2  
...               ...             ...     ...            ...  
25806              99   United-States   <=50K           19.8  
30610              99   United-States    >50K           19.8  
10143              99   United-States   <=50K           19.8  
4308               99   United-States   <=50K           19.8  
1887               99   United-States    >50K           19.8  

[32561 rows x 16 columns]

In [ ]:
salarios.head()

age          workclass  fnlwgt   education  education-num  \
0   39          State-gov   77516   Bachelors             13   
1   50   Self-emp-not-inc   83311   Bachelors             13   
2   38            Private  215646     HS-grad              9   
3   53            Private  234721        11th              7   
4   28            Private  338409   Bachelors             13   

        marital-status          occupation    relationship    race      sex  \
0        Never-married        Adm-clerical   Not-in-family   White     Male   
1   Married-civ-spouse     Exec-managerial         Husband   White     Male   
2             Divorced   Handlers-cleaners   Not-in-family   White     Male   
3   Married-civ-spouse   Handlers-cleaners         Husband   Black     Male   
4   Married-civ-spouse      Prof-specialty            Wife   Black   Female   

   capital-gain  capital-loss  hours-per-week  native-country  salary  \
0          2174             0              40   United-States   <=50K   
1             0             0              13   United-States   <=50K   
2             0             0              40   United-States   <=50K   
3             0             0              40   United-States   <=50K   
4             0             0              40            Cuba   <=50K   

   hours-per-day  
0            8.0  
1            2.6  
2            8.0  
3            8.0  
4            8.0

In [ ]:
salarios.sort_values("native-country", ascending=True)

age          workclass  fnlwgt      education  education-num  \
21541   27            Private  169958        5th-6th              3   
18277   90            Private  311184      Bachelors             13   
27305   34   Self-emp-not-inc  114185      Bachelors             13   
5854    40            Private   37618   Some-college             10   
27299   37   Self-emp-not-inc  327164   Some-college             10   
...    ...                ...     ...            ...            ...   
23814   41   Self-emp-not-inc  145441      Assoc-voc             11   
7287    35            Private  164526        HS-grad              9   
31519   29            Private  273051      Bachelors             13   
4447    25            Private  191230   Some-college             10   
18247   41            Private  324629      Bachelors             13   

            marital-status         occupation    relationship    race  \
21541        Never-married       Craft-repair       Own-child   White   
18277   Married-civ-spouse              Sales         Husband   White   
27305             Divorced   Transport-moving   Not-in-family   White   
5854    Married-civ-spouse       Craft-repair         Husband   White   
27299   Married-civ-spouse       Craft-repair         Husband   White   
...                    ...                ...             ...     ...   
23814   Married-civ-spouse       Craft-repair         Husband   White   
7287    Married-civ-spouse      Other-service         Husband   White   
31519   Married-civ-spouse    Exec-managerial         Husband   White   
4447         Never-married    Exec-managerial       Own-child   White   
18247   Married-civ-spouse    Exec-managerial         Husband   White   

           sex  capital-gain  capital-loss  hours-per-week native-country  \
21541     Male             0             0              40              ?   
18277     Male             0             0              20              ?   
27305     Male             0             0              50              ?   
5854      Male             0             0              40              ?   
27299     Male             0             0              70              ?   
...        ...           ...           ...             ...            ...   
23814     Male             0             0              40     Yugoslavia   
7287      Male             0             0              40     Yugoslavia   
31519     Male             0             0              52     Yugoslavia   
4447    Female             0             0              40     Yugoslavia   
18247     Male             0             0              40     Yugoslavia   

       salary  hours-per-day  
21541   <=50K            8.0  
18277   <=50K            4.0  
27305   <=50K           10.0  
5854    <=50K            8.0  
27299   <=50K           14.0  
...       ...            ...  
23814   <=50K            8.0  
7287     >50K            8.0  
31519    >50K           10.4  
4447    <=50K            8.0  
18247   <=50K            8.0  

[32561 rows x 16 columns]

In [ ]:
salarios.keys()

Index(['age', 'workclass', 'fnlwgt', 'education', 'education-num',
       'marital-status', 'occupation', 'relationship', 'race', 'sex',
       'capital-gain', 'capital-loss', 'hours-per-week', 'native-country',
       'salary', 'hours-per-day'],
      dtype='object')

In [ ]:
lista_ocupaciones=salarios["occupation"].unique()
for valor in lista_ocupaciones:
    print(valor)

 Adm-clerical
 Exec-managerial
 Handlers-cleaners
 Prof-specialty
 Other-service
 Sales
 Craft-repair
 Transport-moving
 Farming-fishing
 Machine-op-inspct
 Tech-support
 ?
 Protective-serv
 Armed-Forces
 Priv-house-serv


In [ ]:
salarios[salarios["sex"]==" Male"]["occupation"]

0              Adm-clerical
1           Exec-managerial
2         Handlers-cleaners
3         Handlers-cleaners
7           Exec-managerial
                ...        
32553          Tech-support
32554       Exec-managerial
32555       Protective-serv
32557     Machine-op-inspct
32559          Adm-clerical
Name: occupation, Length: 21790, dtype: category
Categories (15, object): [' ?', ' Adm-clerical', ' Armed-Forces', ' Craft-repair', ...,
                          ' Protective-serv', ' Sales', ' Tech-support', ' Transport-moving']

In [ ]:
salarios["age"].unique()

array([39, 50, 38, 53, 28, 37, 49, 52, 31, 42, 30, 23, 32, 40, 34, 25, 43,
       54, 35, 59, 56, 19, 20, 45, 22, 48, 21, 24, 57, 44, 41, 29, 18, 47,
       46, 36, 79, 27, 67, 33, 76, 17, 55, 61, 70, 64, 71, 68, 66, 51, 58,
       26, 60, 90, 75, 65, 77, 62, 63, 80, 72, 74, 69, 73, 81, 78, 88, 82,
       83, 84, 85, 86, 87])

In [ ]:
salarios["sex"]=salarios["sex"].replace({" Male": "Male", " Female": "Female"})

In [ ]:
salarios[salarios["sex"]=="Male"]

age          workclass  fnlwgt      education  education-num  \
0       39          State-gov   77516      Bachelors             13   
1       50   Self-emp-not-inc   83311      Bachelors             13   
2       38            Private  215646        HS-grad              9   
3       53            Private  234721           11th              7   
7       52   Self-emp-not-inc  209642        HS-grad              9   
...    ...                ...     ...            ...            ...   
32553   32            Private  116138        Masters             14   
32554   53            Private  321865        Masters             14   
32555   22            Private  310152   Some-college             10   
32557   40            Private  154374        HS-grad              9   
32559   22            Private  201490        HS-grad              9   

            marital-status          occupation    relationship  \
0            Never-married        Adm-clerical   Not-in-family   
1       Married-civ-spouse     Exec-managerial         Husband   
2                 Divorced   Handlers-cleaners   Not-in-family   
3       Married-civ-spouse   Handlers-cleaners         Husband   
7       Married-civ-spouse     Exec-managerial         Husband   
...                    ...                 ...             ...   
32553        Never-married        Tech-support   Not-in-family   
32554   Married-civ-spouse     Exec-managerial         Husband   
32555        Never-married     Protective-serv   Not-in-family   
32557   Married-civ-spouse   Machine-op-inspct         Husband   
32559        Never-married        Adm-clerical       Own-child   

                      race   sex  capital-gain  capital-loss  hours-per-week  \
0                    White  Male          2174             0              40   
1                    White  Male             0             0              13   
2                    White  Male             0             0              40   
3                    Black  Male             0             0              40   
7                    White  Male             0             0              45   
...                    ...   ...           ...           ...             ...   
32553   Asian-Pac-Islander  Male             0             0              11   
32554                White  Male             0             0              40   
32555                White  Male             0             0              40   
32557                White  Male             0             0              40   
32559                White  Male             0             0              20   

       native-country  salary  hours-per-day  
0       United-States   <=50K            8.0  
1       United-States   <=50K            2.6  
2       United-States   <=50K            8.0  
3       United-States   <=50K            8.0  
7       United-States    >50K            9.0  
...               ...     ...            ...  
32553          Taiwan   <=50K            2.2  
32554   United-States    >50K            8.0  
32555   United-States   <=50K            8.0  
32557   United-States    >50K            8.0  
32559   United-States   <=50K            4.0  

[21790 rows x 16 columns]

In [ ]:
lista_ocupaciones=salarios["occupation"].unique()
for valor in lista_ocupaciones:
    print(valor)

 Adm-clerical
 Exec-managerial
 Handlers-cleaners
 Prof-specialty
 Other-service
 Sales
 Craft-repair
 Transport-moving
 Farming-fishing
 Machine-op-inspct
 Tech-support
 ?
 Protective-serv
 Armed-Forces
 Priv-house-serv


In [ ]:
lista_ocupaciones_corregidas=["Adm-clerical", "Exec-managerial", "Handlers-cleaners", "Prof-specialty",
                              "Other-service", "Sales", "Craft-repair", "Transport-moving", "Farming-fishing",
                              "Machine-op-inspct", "Tech-support", "Desconocido", "Protective-serv",
                              "Armed-Forces","Priv-house-serv"]

In [ ]:
print(len(lista_ocupaciones))
print(len(lista_ocupaciones_corregidas))

15
15


In [ ]:
for i in range(0,10):
    print(i)

0
1
2
3
4
5
6
7
8
9


In [ ]:
salarios.replace({lista_ocupaciones[0]:lista_ocupaciones_corregidas[0]})

age          workclass  fnlwgt    education  education-num  \
0       39          State-gov   77516    Bachelors             13   
1       50   Self-emp-not-inc   83311    Bachelors             13   
2       38            Private  215646      HS-grad              9   
3       53            Private  234721         11th              7   
4       28            Private  338409    Bachelors             13   
...    ...                ...     ...          ...            ...   
32556   27            Private  257302   Assoc-acdm             12   
32557   40            Private  154374      HS-grad              9   
32558   58            Private  151910      HS-grad              9   
32559   22            Private  201490      HS-grad              9   
32560   52       Self-emp-inc  287927      HS-grad              9   

            marital-status          occupation    relationship    race  \
0            Never-married        Adm-clerical   Not-in-family   White   
1       Married-civ-spouse     Exec-managerial         Husband   White   
2                 Divorced   Handlers-cleaners   Not-in-family   White   
3       Married-civ-spouse   Handlers-cleaners         Husband   Black   
4       Married-civ-spouse      Prof-specialty            Wife   Black   
...                    ...                 ...             ...     ...   
32556   Married-civ-spouse        Tech-support            Wife   White   
32557   Married-civ-spouse   Machine-op-inspct         Husband   White   
32558              Widowed        Adm-clerical       Unmarried   White   
32559        Never-married        Adm-clerical       Own-child   White   
32560   Married-civ-spouse     Exec-managerial            Wife   White   

          sex  capital-gain  capital-loss  hours-per-week  native-country  \
0        Male          2174             0              40   United-States   
1        Male             0             0              13   United-States   
2        Male             0             0              40   United-States   
3        Male             0             0              40   United-States   
4      Female             0             0              40            Cuba   
...       ...           ...           ...             ...             ...   
32556  Female             0             0              38   United-States   
32557    Male             0             0              40   United-States   
32558  Female             0             0              40   United-States   
32559    Male             0             0              20   United-States   
32560  Female         15024             0              40   United-States   

       salary  hours-per-day  
0       <=50K            8.0  
1       <=50K            2.6  
2       <=50K            8.0  
3       <=50K            8.0  
4       <=50K            8.0  
...       ...            ...  
32556   <=50K            7.6  
32557    >50K            8.0  
32558   <=50K            8.0  
32559   <=50K            4.0  
32560    >50K            8.0  

[32561 rows x 16 columns]

In [ ]:
for i in range(0,15):
    salarios=salarios.replace({lista_ocupaciones[i]:lista_ocupaciones_corregidas[i]})

In [ ]:
lista_ocupaciones=salarios["occupation"].unique()
for valor in lista_ocupaciones:
    print(valor)

Adm-clerical
Exec-managerial
Handlers-cleaners
Prof-specialty
Other-service
Sales
Craft-repair
Transport-moving
Farming-fishing
Machine-op-inspct
Tech-support
Desconocido
Protective-serv
Armed-Forces
Priv-house-serv


Modificar la variable salary de modo que los valores "<=50K" sean 0s y ">50" sean 1s

In [ ]:
salarios["salary"]=salarios["salary"].replace({" <=50K": 0, " >50K": 1})
salarios.head()

age          workclass  fnlwgt   education  education-num  \
0   39          State-gov   77516   Bachelors             13   
1   50   Self-emp-not-inc   83311   Bachelors             13   
2   38            Private  215646     HS-grad              9   
3   53            Private  234721        11th              7   
4   28            Private  338409   Bachelors             13   

        marital-status         occupation    relationship    race     sex  \
0        Never-married       Adm-clerical   Not-in-family   White    Male   
1   Married-civ-spouse    Exec-managerial         Husband   White    Male   
2             Divorced  Handlers-cleaners   Not-in-family   White    Male   
3   Married-civ-spouse  Handlers-cleaners         Husband   Black    Male   
4   Married-civ-spouse     Prof-specialty            Wife   Black  Female   

   capital-gain  capital-loss  hours-per-week  native-country salary  \
0          2174             0              40   United-States      0   
1             0             0              13   United-States      0   
2             0             0              40   United-States      0   
3             0             0              40   United-States      0   
4             0             0              40            Cuba      0   

   hours-per-day  
0            8.0  
1            2.6  
2            8.0  
3            8.0  
4            8.0


1.   ¿Cuál es la edad promedio de los empleados?
2.   ¿En qué porcentajes se distribuyen las diferentes raza?
3. De las personas que ganan más de 50 mil, ¿cuántas son mujeres?
4. Analizar la relación entre el nivel de estudios y los sueldos




In [ ]:
promedio_edades=salarios["age"].mean()
print("La edad promedio de los empleados es de aproximadamente", round(promedio_edades), "años")

La edad promedio de los empleados es de aproximadamente 39 años
